core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -62,12 +62,15 @@ public BoundedRangeFileInputStream(FSDataInputStream in, long offset, long lengt
 
   @Override
   public int available() throws IOException {
-    int avail = in.available();
-    if (pos + avail > end) {
-      avail = (int) (end - pos);
-    }
+    final FSDataInputStream inLocal = in;
+    synchronized (inLocal) {
+      int avail = inLocal.available();
```

Comment:
> I am considering this, but others had commented to me that adhering to the API as close as possible was a good thing. 

Thats a hard call to make, I don't know what the right answer is.  We can make BoundedRangeInputStream work with the code that currently calls it.  To make it work with code that might call it in the future, need to follow the interface javadoc.  For example the way hadoop uses it could change in the future.

Calling `in.seek()` then `in.available()` seems like the correct thing to do.  However, based on what @phrocker  said, `in.seek()` call may have a performance penalty, with no real benefit.  I don't really like that.   One thing I am thinking is that always returning `end-pos` seems more correct than only calling `in.available()` w/o calling `in.seek()`.